File: yaazh_admin/test/jpeg_jfif_test.dart
import 'dart:typed_data';

import 'package:flutter_test/flutter_test.dart';
import 'package:yaazh_admin/core/jpeg_jfif.dart';

void main() {
  test('strips EXIF APP1 and inserts JFIF', () {
    final input = Uint8List.fromList([
      0xFF, 0xD8,
      0xFF, 0xE1, 0x00, 0x06, 0x45, 0x78, 0x00, 0x00,
      0xFF, 0xD9,
    ]);
    final out = toJfifJpeg(input);
    expect(isJpegMagic(out), isTrue);
    expect(out[2], 0xFF);
    expect(out[3], 0xE0);
    expect(out.sublist(out.length - 2), [0xFF, 0xD9]);
    var hasExif = false;
    for (var i = 2; i < out.length - 1; i++) {
      if (out[i] == 0xFF && out[i + 1] == 0xE1) hasExif = true;
    }
    expect(hasExif, isFalse);
  });
}

File: backend-old/tests/unit/jpeg-jfif.test.ts
import { describe, expect, it } from "vitest";
import { toJfifJpeg } from "../../src/utils/jpeg-jfif.js";

describe("toJfifJpeg", () => {
  it("strips APP1 EXIF and inserts JFIF", () => {
    const input = Buffer.from([
      0xff, 0xd8, 0xff, 0xe1, 0x00, 0x06, 0x45, 0x78, 0x00, 0x00, 0xff, 0xd9,
    ]);
    const out = toJfifJpeg(input);
    expect(out[0]).toBe(0xff);
    expect(out[1]).toBe(0xd8);
    expect(out[2]).toBe(0xff);
    expect(out[3]).toBe(0xe0);
    expect(out.includes(Buffer.from([0xff, 0xe1]))).toBe(false);
  });
});
